### ISSUE TITLE
LEDs stay on when turning off Raspberry Pi - SOLUTION!

### ISSUE THREAD

**Issue by cmvac** [hyperion-project/hyperion#575]
I noticed since I updated to the last Hyperion version, whenever I turned off my Pi, the LEDs wouldn't turn off. I found a solution for that, maybe it could be included in the main Hyperion code.

I modified the content of hyperion.init.sh and added a line in:

stop)
        if [ -f $PIDFILE ]
        then
			/storage/hyperion/bin/hyperion-remote.sh -c black 
			    sleep 1

Here is my full file: http://pastebin.com/fNFaZd5N

I don't know if anybody else is having this issue, but if so, this is the solution. 

Have a nice day! :)

**Comment by brindosch** [comment]
i imagine i did that, hm it seams lost during -unknown-
This is just a problem for init.d user (wheezy and other distris that uses init.d)
If you upgrade to jessie later that will work out of the box
I will include it - again ;)

Question - you are using OpenELEC? Now i am confused, this should not work, casue that script is not used for OE?!

**Comment by cmvac** [comment]
Yes, I am using OpenELEC and it did solve my issue. ;)  To be honest, I didn't know that, but the fact remains that the LEDs now turn off when shutting down. Weird...

**Comment by brindosch** [comment]
OpenELEC just triggers your autostart.sh and shutdown.sh . Do you have a shutdown.sh (it is not created by default)?

**Comment by cmvac** [comment]
Is the shutdown.sh file supposed to be inside the .config folder? Because if so, I don't have it. :/ Nor in the storage folder or the userdata folder.

**Comment by cmvac** [comment]
It is supposed to be in here, right? 

/storage/.config/shutdown.sh

I have reinstalled my OpenELEC system two weeks ago, but never noticed that file before. I did this change in the hyperion.init.sh file before, on my previous image of OpenELEC, in order to get the LEDs shutting down as well. Then I forgot about it, because it was working fine. Now I had to do it again, and decided to post it here.

**Comment by penfold42** [comment]
We should make sure hyperiond sets the LEDs to off when it gets a kill signal.

I suspect not all led drivers do the right thing here

**Comment by brindosch** [comment]
I tested these some weeks ago with my apa102. If i use a systemd driven service. It is working. If i use openelc and i tested very much things. I can´t send a -black signal fast enough to hyperion inside shutdown.sh
the point is now: some people can use it! So i investigated why that happend.

I logged the output - hyperion is not reachable in just this moment - it is already killed
that means: the longer the system needs to shutdown (bloated with much servies and use a Pi1 rather than a Pi2/Pi3 (perfomance)) the higher is the chance to get that working
This is my attempt to explain it.

@cmvac 
It doesn´t make sense, i can´t understand this. Just the idea above
